File: app/src/main/java/me/iacn/biliroaming/StreamUtils.java
package me.iacn.biliroaming;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.zip.GZIPInputStream;

/**
 * Created by iAcn on 2019/3/26
 * Email i@iacn.me
 */
public class StreamUtils {

    public static String getContent(InputStream inputStream, String encoding) {
        String result = null;
        ByteArrayOutputStream byteArrayStream = null;

        try {
            if ("gzip".equalsIgnoreCase(encoding))
                inputStream = new GZIPInputStream(inputStream);

            byte[] buffer = new byte[2048];
            byteArrayStream = new ByteArrayOutputStream();
            for (int len; (len = inputStream.read(buffer)) > 0; ) {
                byteArrayStream.write(buffer, 0, len);
            }

            byteArrayStream.flush();
            result = new String(byteArrayStream.toByteArray());

        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                if (inputStream != null) inputStream.close();

                if (byteArrayStream != null) byteArrayStream.close();

            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        return result;
    }
}

File: app/src/main/java/me/iacn/biliroaming/network/StreamUtils.kt
package me.iacn.biliroaming.network

import java.io.ByteArrayOutputStream
import java.io.IOException
import java.io.InputStream
import java.util.zip.GZIPInputStream

/**
 * Created by iAcn on 2019/3/26
 * Email i@iacn.me
 */
object StreamUtils {
    @JvmStatic
    fun getContent(input: InputStream?, encoding: String?): String? {
        var inputStream = input;
        var result: String? = null
        var byteArrayStream: ByteArrayOutputStream? = null
        try {
            if ("gzip".equals(encoding, ignoreCase = true)) inputStream = GZIPInputStream(inputStream)
            val buffer = ByteArray(2048)
            byteArrayStream = ByteArrayOutputStream()
            var len: Int
            while (inputStream!!.read(buffer).also { len = it } > 0) {
                byteArrayStream.write(buffer, 0, len)
            }
            byteArrayStream.flush()
            result = String(byteArrayStream.toByteArray())
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            try {
                inputStream?.close()
                byteArrayStream?.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
        return result
    }
}